needy/library.py

```diff
@@ -176,16 +173,21 @@ def __write_build_status(self):
             json.dump(status, status_file)
 
     def __cache_artifacts(self):
-        if self.__build_cache:
+        if self.__build_caches:
             try:
-                self.__build_cache.store_artifacts(self.build_directory(), self.__cache_key())
+                self.__build_caches[0].store_artifacts(self.build_directory(), self.__cache_key())
```

Comment:
This looks fishy. We just pick the first build cache and store the artifacts there?